[#1178] Make github profile and repo hyperlinks more consistent

ccyccyccy opened the pull request:
Fixes #1178 and  #1191.
```
When grouped by repo, the same icon to link to the repo is duplicated
multiple times for no reason. Also, the icon to link to the author's
profile is hard to find.

We can make the links more intuitive and consistent by putting links to
the author's profile beside author names, and links to repo beside repo
names.

Let's reorder the icon group such that the hyperlinks are closer to the
respective names, as well as always showing the icons beside their
respective names. As this change heavily relies on the icons being
responsive, let's resolve issue #1191 as well to make use of
vue-fontawesome to make the icons responsive.
```

Note that there are some existing bugs that affect this PR.
1. Affected by #1185, but should work with the previews as the `config.json` is specified. Preferably merge after #1185 is resolved.

anubh-v commented on frontend/src/summary.pug:
```diff
@@ -154,23 +158,34 @@
           ) {{ user.displayName }} ({{ user.name }})
           .summary-chart__title--contribution.mini-font [{{ user.totalCommits }} lines]
+          a(
+            v-if="filterGroupSelection !== 'groupByRepos'"
```
I think you're missing a comma at the end of the line, which is causing the errors from pug-lint
Here is more info on the error: [link](https://github.com/pugjs/pug-lint/blob/master/docs/rules.md#validateattributeseparator-string--object)

ccyccyccy replied:
Oops I forgot to check the travis result. Thanks for the catch :)

jamessspanggg commented:
One qn: what would be the downsides if we were to use [vue fontawesome icons](https://github.com/FortAwesome/vue-fontawesome), which alleviates the need for the span workaround?

ccyccyccy commented:
> 
> 
> One qn: what would be the downsides if we were to use [vue fontawesome icons](https://github.com/FortAwesome/vue-fontawesome), which alleviates the need for the span workaround?

Depends on how the library works, but we'll need to remember to remove the useless spans after using the new library.

jamessspanggg commented:
> Depends on how the library works, but we'll need to remember to remove the useless spans after using the new library.

I was thinking whether we should just switch all the icons to vue fontawesome, in this way we do not need to add and remove again. @eugenepeh @fzdy1914 may I get ur opinion on this?

fzdy1914 commented:
I prefer to use vue fontawesome directly.

eugenepeh commented:
>I was thinking whether we should just switch all the icons to vue fontawesome, in this way we do not need to add and remove again. @eugenepeh @fzdy1914 may I get ur opinion on this?

Sounds good to me.

ccyccyccy commented:
I changed to use vue-fontawesome for all the icons in the repo in this PR. Should I split it to another PR? I think it's good to merge it into this PR since this PR heavily relies on the icons being reactive.

jylee-git commented:
Functionality seems to be fine.

Resolve the conflicts and request for review again

ccyccyccy commented:
> Functionality seems to be fine.
> 
> Resolve the conflicts and request for review again

Fixed.
